### ISSUE TITLE
com.pushwoosh.thirdpart.com.ironz.binaryprefs.j.a.b deserialization crash in 5.4.1

### ISSUE THREAD

**Issue by n-belokopytov** [Pushwoosh/pushwoosh-android-sdk#81]
com.pushwoosh.thirdpart.com.ironz.binaryprefs.d.a: 
  at com.pushwoosh.thirdpart.com.ironz.binaryprefs.j.a.b (Unknown Source)
  at com.pushwoosh.thirdpart.com.ironz.binaryprefs.f.c.b (Unknown Source)
  at com.pushwoosh.thirdpart.com.ironz.binaryprefs.f.c.a (Unknown Source)
  at com.pushwoosh.thirdpart.com.ironz.binaryprefs.a.getAll (Unknown Source)
  at com.pushwoosh.internal.utils.h.a (Unknown Source)
  at com.pushwoosh.a.f.a (Unknown Source)
  at com.pushwoosh.notification.e.e (Unknown Source)
  at com.pushwoosh.e.a (Unknown Source)
  at com.pushwoosh.e.a (Unknown Source)
  at com.pushwoosh.h.onReceive (Unknown Source)
  at com.pushwoosh.internal.event.EventBus.b (Unknown Source)
  at com.pushwoosh.internal.event.EventBus.c (Unknown Source)
  at com.pushwoosh.internal.event.EventBus.a (Unknown Source)
  at com.pushwoosh.internal.event.c.run (Unknown Source)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6123)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:867)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:757)

**Comment by n-belokopytov** [comment]
Here's more. Please figure out your third party solution for serialization.


```
com.pushwoosh.thirdpart.com.ironz.binaryprefs.j.a.b
java.lang.UnsupportedClassVersionError: Flag verification failed. Incorrect flag '0'
```

```
com.pushwoosh.thirdpart.com.ironz.binaryprefs.j.a.b
com.pushwoosh.thirdpart.com.ironz.binaryprefs.d.d: pushwoosh_local_push_bundle_573$##$title_header key's value is zero bytes for deserialize
```

```
com.pushwoosh.thirdpart.com.ironz.binaryprefs.j.a.b
com.pushwoosh.thirdpart.com.ironz.binaryprefs.d.d: pushwoosh_local_push_bundle_257$##$local_push_id key's value is zero bytes for deserialize
```

```
com.pushwoosh.thirdpart.com.ironz.binaryprefs.j.a.b
com.pushwoosh.thirdpart.com.ironz.binaryprefs.d.d: pushwoosh_local_push_bundle_227$##$ci key's value is zero bytes for deserialize
```

```
com.pushwoosh.thirdpart.com.ironz.binaryprefs.j.a.b
com.pushwoosh.thirdpart.com.ironz.binaryprefs.d.d: pushwoosh_local_push_bundle_197$##$pw_msg key's value is zero bytes for deserialize
```

```
com.pushwoosh.thirdpart.com.ironz.binaryprefs.j.a.b
com.pushwoosh.thirdpart.com.ironz.binaryprefs.d.d: pushwoosh_local_push_bundle_434$##$type key's value is zero bytes for deserialize
```

**Comment by shromyak** [comment]
Same for me 
<img width="1112" alt="screen shot 2017-11-12 at 10 45 01 pm" src="https://user-images.githubusercontent.com/20333321/32703221-3d1f8148-c7fb-11e7-8175-e6e9fafc97a0.png">

**Comment by shromyak** [comment]
private final SharedPreferences a = AndroidPlatformModule.getPrefsProvider().providePrefs("com.pushwoosh.pushnotifications.cached_requests");
As temporary remedy will try to intercept this stupid prefs to get rid of broken data.
I'm very disappointed that Pushwoosh treats their clients as laboratory animals to test experimental solutions such as Binary Preferences. It's second f..up within last month related with "smooth upgrade", I haven't such trouble with Urban Airship :(

**Comment by anry200** [comment]
Same issue here

**Comment by shaders** [comment]
We are working on a fix to be released asap.

**Comment by wfhm** [comment]
@n-belokopytov @shromyak @anry200 

The issue related to Binary Preferences library is fixed with 5.4.3. 

@n-belokopytov The Pushwoosh storage is rolled back to v5.1.1, so OOM exceptions will return. We are working on a better solution right now.

**Comment by shromyak** [comment]
Yes, and now you screwed-up legacy GCM messaging, thank you, nice fix!

**Comment by iamironz** [comment]
@shromyak may you try to publish an issue on my repository instead of passive-agressive position?

**Comment by iamironz** [comment]
Also I wanna know what version of binaryprefs do you use for sdk implementation? 
As can I see you use old serialization protocol which is implemented inside very old version on my library.

**Comment by gholz** [comment]
@wfhm Does the 5.5.0 version solves the OOM exceptions?

**Comment by wfhm** [comment]
@gholz 5.5.0 does not include this fix. It requires storage refactoring, and will be rolled out in upcoming releases.

**Comment by wfhm** [comment]
@n-belokopytov OOM exceptions are fixed in 5.5.9.
